### PR TITLE
Set up autodeploy with circle

### PR DESCRIPTION
**Pull request by pimterry**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+version: 2
+
+buildSteps: &buildSteps
+  - checkout
+  - restore_cache:
+      keys:
+        - dependency-cache-{{ checksum "package.json" }}
```
Comment (brownjohnf) [inline circle.yml:8]: I don't see you setting this key anywhere. Am I missing something about how circle caching works?

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      - deploy:
+          # Deploy passing builds if they're tagged with a version
+          requires:
+            - "node-6"
+            - "node-8"
+          filters:
+            tags:
+              only: /^v\d+\.\d+\.\d+$/
+            branches:
+              ignore: /.*/
```
Comment (brownjohnf) [inline circle.yml:61]: Is it worth also restricting this to master, for safety?

**Review by brownjohnf — COMMENTED** [review]
I'll have to get the token shared with me, and then I'll add it.

File: circle.yml (hunk 1 above)
Comment (pimterry) [reply circle.yml:8]: You're right, I'd mixed up two examples and ended up with mismatched keys. This is supposed to be set in the `save_cache` step below. Now fixed (I think).

File: circle.yml (hunk 2 above)
Comment (pimterry) [reply circle.yml:61]: You can't, sadly. Git tags aren't explicitly related to branches, they're independent standalone references. This line is just ignoring every single branch, so it doesn't get deployed for master _and_ the tagged commit.

What we could try to say to get that sort of behaviour is "does this tag refer to a commit that's also on master", but that's the best you can theoretically get in terms of relating the two, and circle (or travis or anything else) doesn't seem to have any kind of built-in support for filtering like that, so it's a little tricky to do. I'd like to extend to do that kind of thing eventually, but I don't _think_ it's strictly necessary for a first implementation. I can look into it more if you disagree though.

File: circle.yml (hunk 1 above)
Comment (pimterry) [reply circle.yml:8]: Yep, tested, it seems to now be happily picking up this cache: https://circleci.com/gh/resin-io-modules/resin-pine/34

**Comment by brownjohnf** [comment]
Okay. Looks good; I'll approve when the token is set up.

**Review by brownjohnf — APPROVED** [review]
Okay, token should be set up. Since this only runs on versioned builds, in theory we could enable forked builds for this repo. Let me know if you want me to do that.

**Pull request merged**